aws/resource_aws_api_gateway_rest_api.go

```diff
@@ -335,6 +343,30 @@ func resourceAwsApiGatewayRestApiUpdateOperations(d *schema.ResourceData) []*api
 		}
 	}
 
+	if d.HasChange("endpoint_configuration.1.vpc_endpoint_ids") {
+		o, n := d.GetChange("endpoint_configuration.1.vpc_endpoint_ids")
+		prefix := "/endpointConfiguration/vpc_endpoint_ids"
+
+		old := o.([]interface{})
+		new := n.([]interface{})
+
+		for _, v := range old {
+			operations = append(operations, &apigateway.PatchOperation{
+				Op:   aws.String("remove"),
+				Path: aws.String(fmt.Sprintf("%s/%s", prefix, escapeJsonPointer(v.(string)))),
+			})
+		}
+
+		if len(new) > 0 {
+			for _, v := range new {
+				operations = append(operations, &apigateway.PatchOperation{
+					Op:   aws.String("add"),
+					Path: aws.String(fmt.Sprintf("%s/%s", prefix, escapeJsonPointer(v.(string)))),
```

Comment:
This doesn't appear to work for a few reasons:
- The path should be `/endpointConfiguration/vpcEndpointIds` according to https://docs.aws.amazon.com/apigateway/latest/developerguide/associate-private-api-with-vpc-endpoint.html
- Should use `endpoint_configuration.0.` not `.1`

Here's an alternative implementation. Tested in a real deployment.

```go
	if d.HasChange("endpoint_configuration.0.vpc_endpoint_ids") {
		o, n := d.GetChange("endpoint_configuration.0.vpc_endpoint_ids")
		prefix := "/endpointConfiguration/vpcEndpointIds"

		old := o.([]interface{})
		new := n.([]interface{})

		for _, v := range old {
			operations = append(operations, &apigateway.PatchOperation{
				Op:    aws.String("remove"),
				Path:  aws.String(prefix),
				Value: aws.String(v.(string)),
			})
		}

		for _, v := range new {
			operations = append(operations, &apigateway.PatchOperation{
				Op:    aws.String("add"),
				Path:  aws.String(prefix),
				Value: aws.String(v.(string)),
			})
		}
	}
```

Reply:
fixed, thanks!